site/app/views/submission/HomeworkView.php

```diff
@@ -445,21 +445,38 @@ public function showGradeable($gradeable, $gradeable_select, $days_late) {
 HTML;
                             }
                             else {
-                                $background = "";
-                                if ($testcase->getPointsAwarded() >= $testcase->getPoints()) {
-                                    $background = "green-background";
+                              $background = "";
+                              if ($testcase->isExtraCredit()) {
+                                if ($testcase->getPointsAwarded() > 0) {
+                                  $background = "green-background";
+                                  $return .= <<<HTML
+                <span class="badge {$background}"> &nbsp; +{$testcase->getPointsAwarded()} &nbsp; </span>
+HTML;
```

Comment:
Yes, it is intentional to not have a badge if you don't earn the extra credit.
Yes, it is intentional to not put how many points are possible.
I feel strongly about both these decisions based on my use of extra credit and the appearance when I tested sample (what I thought were typical) submissions.
I'd like to leave this.  (can be re-discussed in future).